bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/handler/BoschIndegoHandler.java

```diff
@@ -120,6 +140,23 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private void handleRefreshCommand(String channelId) {
+        switch (channelId) {
+            case STATE:
+            case TEXTUAL_STATE:
+            case MOWED:
+            case ERRORCODE:
+            case STATECODE:
+            case READY:
+                scheduler.submit(() -> this.refreshStateWithExceptionHandling());
```

Comment:
I see that it was already done like that before, but can you please explain why it is done asynchronously ?

Reply:
Actually I can't, you have a point. I was somehow assuming it had to be async, probably because I had the hdpowerview binding in mind. I will need to revisit that. Changed to send requests synchronously.